modules/http/tests/http_split_response.cc

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2000-2015 BalaBit IT Ltd, Budapest, Hungary
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ ***************************************************************************/
+
+#define BOOST_TEST_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include "../http.h"
+
+
+extern const gchar *last_log;
+
+HttpProxy* new_proxy() {
+	HttpProxy *proxyFake = (HttpProxy *)malloc(sizeof(HttpProxy));
+	const char *session_id = "test_session";
+	ZProxy *self=(ZProxy *)proxyFake;
+	g_strlcpy(self->session_id, session_id , strlen(session_id)+1);
+
+	proxyFake->request_url=g_string_new(NULL);
+	proxyFake->request_method=g_string_new(NULL);
+	proxyFake->response_msg=g_string_new(NULL);
+	proxyFake->max_url_length=2048;
+
+	return proxyFake;
+}
+
+typedef struct {
+	const char* input_line;
+	const char* expected_response_version;
+	const char* expected_response;
+	gint expected_response_code;
+	const char* expected_response_msg;
+} TestCase;
+
+#define StringOfLength256\
+		"Aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"\
+		"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab"
+
+#define longResponse "HTTP/1.0 200 " StringOfLength256 "a"
+TestCase testCases[] = {
+		{ "HTTP/1.1 200 OK", "HTTP/1.1", "200", 200, "OK" },
+		{ "HTTP/1.1 404 Not Found", "HTTP/1.1", "404", 404, "Not Found" },
+		{ "HTTP/1.1 400 Bad Request", "HTTP/1.1", "400", 400, "Bad Request" },
+		{ longResponse, "HTTP/1.0", "200", 200, StringOfLength256},
+		{NULL, NULL,NULL, 0, NULL}
+};
+
+BOOST_AUTO_TEST_CASE(test_correct_response_is_parsed_correctly)
```

Comment:
Isn't there some mechanism in the test framework for parametrized test cases? It would be nice if we didn't have to surround similar tests with the do-if-break-while boilerplate. Maybe some macro magic?